tests/protocol/test_goal_state.py

```diff
@@ -16,7 +21,63 @@ def test_fetch_goal_state_should_raise_on_incomplete_goal_state(self):
             protocol.mock_wire_data.set_incarnation(2)
 
             with patch('time.sleep') as mock_sleep:
-                with self.assertRaises(IncompleteGoalStateError):
+                with self.assertRaises(ProtocolError):
                     GoalState(protocol.client)
-                self.assertEqual(_NUM_GS_FETCH_RETRIES, mock_sleep.call_count, "Unexpected number of retries")
+                self.assertEqual(_GET_GOAL_STATE_MAX_ATTEMPTS, mock_sleep.call_count, "Unexpected number of retries")
+
+    def test_fetch_full_goal_state_should_save_goal_state_to_history_directory(self):
```

Comment:
This replaces the former test_update_goal_state_should_archive_last_goal_state and test_update_goal_state_should_not_persist_the_protected_settings that used to be in test_wire.py